src/Credentials/EcsCredentialProvider.php

```diff
@@ -86,6 +87,19 @@ public function __invoke()
                         ]
                     )->then(function (ResponseInterface $response) {
                         $result = $this->decodeResult((string)$response->getBody());
+                        if (!isset($result['AccountId']) && isset($result['RoleArn'])) {
+                            try {
+                                $parsedArn = new Arn($result['RoleArn']);
+                                $result['AccountId'] = $parsedArn->getAccountId();
+                            } catch (\Exception $e) {
+                                // AccountId will be null
+                                trigger_error(
```

Comment:
Do users have any control over whether or not an arn is returned and parsed successfully?  Some error handlers may convert this into an exception.

Reply:
I think users should attach a task role to the task definition in order to have a RoleArn returned. Maybe I should avoid this warning and just leave AccountId null when it is not resolve from any of the sources. What do you think?

Reply:
Warning have been removed.